File: 2658/2658.go
func maximumNumberOfFishInAGrid(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := maximumNumberOfFishInAGrid(root.Left)
	r := maximumNumberOfFishInAGrid(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 2658/2658.py
class Solution:
    def maximum_number_of_fish_in_a_grid(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.maximum_number_of_fish_in_a_grid(root.left), self.maximum_number_of_fish_in_a_grid(root.right))